formats/capabilities/SPEC_1.0.md

```diff
@@ -0,0 +1,91 @@
+# Capability format
+
+<dl>
+	<dt>Namespace</dt><dd>`capabilities`</dd>
+	<dt>Version</dt><dd>1.0</dd>
+</dl>
+
+This is an application format for continuous sharing and minting of capability information
+
+## Subspace
+
+Capabilities are by an entity, and are therefore published under the issuing entities subspace.
+
+## Public capabilities
+
+The knowledge that you've issued capabilities to the receiver will be public for all
+
+`/capabilities/1.0/public/{receiver-pubkey}/{id}`
+
+## Private Capabilities
+
+You may want to publish these entries to all (e.g. You'd like help from everyone in the network to distribute, because you don't have a direct peer with the recipient),
+while not allowing people to know who you're actually granting these to.
+
+More formally, given the issuer, a path a capability is published under, and a proposed receiver, it should not be possible for anyone else to know if that is the correct receiver.
+
+For such cases instead of publishing these under the pubkey of the receiving entity, they are instead published under `encrypt(receiver, concat(receiver-pubkey, nonce))`. The receiver must attempt to decrypt all paths until they find their one. The issuer should save and use the same nonce in future so this only needs to happen once.
```

Comment:
Ok, that's super neat, thanks :D

Reply:
Question - does this pose an issue if any other app tries to use the public/secret combination as a shared secret, e.g. for encryption?

Just been reading a bunch of random stuff like [this](https://crypto.stackexchange.com/questions/9418/how-random-is-the-shared-secret-in-the-diffie-hellman-key-agreement/9428#9428)

Perhaps this should be `HKDF-Expand(scalarmult(my_sk, their_pk), "capability-publication", 32)`, so we never disclose the original shared secret?

Reply:
There are other issues as well: if you and I both write capabilities using the naive implementation, we both use the same curve point, so an observer actually sees that the two of us have exchanged capabilities. So there needs to be some further symmetry breaking. So see this "counterproposal" more as opening up some design space rather than as a serious suggestion.

Reply:
But a definite "yes" to "Does this pose an issue if any other app tries to use the public/secret combination as a shared secret, e.g. for encryption?"